ambry-clustermap/src/main/java/com/github/ambry/clustermap/CompositeClusterManager.java

```diff
@@ -34,48 +36,51 @@
  * instance and uses the {@link StaticClusterManager} as the source-of-truth. It relays events to both and checks for
  * and reports inconsistencies in the views from the two underlying cluster managers.
  */
-class CompositeClusterManager implements ClusterMap {
+public class CompositeClusterManager implements ClusterMap {
```

Comment:
it seems like you only need CompositeClusterManager to carry HelixClusterManager and StaticClusterMap to AmbryServer, you don't really care about if the CompositeClusterManager would compare the results from both ClusterMap. If this is the case, we might as well leave CompositeClusterManager intact and only expose the methods to return two clustermaps.

Reply:
I started with that, exposing just the two clustermaps. But that was not enough.

Deep in the code, BlobId::L302 calls clusterMap.getPartitionIdFromStream. The current composite mgr defaults to returning the result from static-cluster-map and this does not fly on the verification node because it relies on helix-cluster-map to discover replicas and extract partitionId from blobIDs.  So I had to change CompositeClusterManager::getPartitionIdFromStream.

Next, HelixClusterManagerTest::basicInterfaceTest fails because it calls clusterManager.getWritablePartitionIds and clusterManager.getPartitionIdFromStream. The types of return values don't match. So then I had to change other methods.

Finally, I made the checking of val from static and helix maps optional because the result will always be false on the verification node. The check is not needed there.